Code/Framework/AzToolsFramework/Tests/InstanceDataHierarchy.cpp

```diff
@@ -1315,4 +1463,108 @@ namespace UnitTest
+    {
+        using AzToolsFramework::InstanceDataHierarchy;
+        using AzToolsFramework::InstanceDataNode;
+
+        for (auto child : componentNode1->GetChildren())
+        {
+            AZStd::string childName(child.GetElementMetadata()->m_name);
+            if (childName.compare("GroupFloat") == 0)
+            {
+                EXPECT_EQ(child.GetGroupElementMetadata()->m_description, "Normal Group");
+            }
+            if (childName.compare("ToggleGroupInt") == 0)
+            {
+                EXPECT_EQ(child.GetGroupElementMetadata()->m_description, "Group Toggle");
+            }
+            if ((childName.compare("SubDataNormal") == 0) || (childName.compare("SubDataToggle") == 0))
+            {
+                for (auto subChild : child.GetChildren())
+                {
+                    childName = subChild.GetElementMetadata()->m_name;
+                    if (childName.compare("SubInt") == 0)
+                    {
+                        EXPECT_EQ(subChild.GetGroupElementMetadata()->m_description, "Normal SubGroup");
+                    }
+                    if (childName.compare("SubFloat") == 0)
+                    {
+                        EXPECT_EQ(subChild.GetGroupElementMetadata()->m_description, "SubGroup Toggle");
+                    }
+                }
+            }
+        }
+    }
+
+    class InstanceDataHierarchyGroupTestFixtureParameterized
+        : public InstanceDataHierarchyGroupTestFixture
+        , public ::testing::WithParamInterface<const char*>
+    {
+    };
+
+    INSTANTIATE_TEST_CASE_P(
+        InstanceDataHierarchyGroupTestFixture,
+        InstanceDataHierarchyGroupTestFixtureParameterized,
+        ::testing::Values("GroupFloat", "GroupToggle", "ToggleGroupInt", "SubInt", "SubToggle", "SubFloat"));
+
+    // Test to validate that each node in a group and Subgroup has the correct parent
+    TEST_P(InstanceDataHierarchyGroupTestFixtureParameterized, ValidatingGroupAndSubGroupParents)
+    {
+        using AzToolsFramework::InstanceDataHierarchy;
+        using AzToolsFramework::InstanceDataNode;
+
+        const char* paramName = GetParam();
+        for (auto child : componentNode1->GetChildren())
+        {
+            AZStd::string childName(child.GetElementMetadata()->m_name);
+            if (childName.compare(paramName) == 0)
+            {
+                EXPECT_EQ(child.GetParent()->GetClassMetadata()->m_name, "GroupTestComponent");
+            }
+            if ((childName.compare("SubDataNormal") == 0) || (childName.compare("SubDataToggle") == 0))
+            {
+                for (auto subChild : child.GetChildren())
+                {
+                    childName = subChild.GetElementMetadata()->m_name;
+                    if (childName.compare(paramName) == 0)
+                    {
+                        EXPECT_EQ(subChild.GetParent()->GetClassMetadata()->m_name, "SubData");
```

Comment:
question: I just want to double check here EXPECT_EQ is doing a proper string compare - do you know what this outputs when it fails? (if you update `SubData` to `SubDate` for example just to make sure it fails) - you might be better off using one of the GTest str test functions (though they might only work with std::string types potentially). It's just if you use them you will likely get much better output when the test fails. Worth trying just to be sure but looking much better now

Reply:
Yeah! When the test fails, it prints out the input, the output and the expected value. I found that information very helpful, since if any of these tests were to fail, the developer would have information about exactly which input the test was failing on. This is especially helpful when only some inputs fail. For example, if I change that subChild test to SubDate, then I'll see each of the failing outputs. I'll also see that each of the failing inputs belongs to a sub group, so I'll know to look at the parenting of sub groups.

Reply:
That's awesome! 😄 Thanks for the info 👍